src/time.rs

```diff
@@ -0,0 +1,141 @@
+use std::time::Duration;
+
+pub fn format_duration(duration: &Duration) -> String {
+    let raw_seconds = duration.as_secs();
+
+    let hours = raw_seconds / 3600;
+    let minutes = (raw_seconds % 3600) / 60;
+    let seconds = raw_seconds % 60;
+
+    let hours_label = match hours {
+        1 => "hour",
+        _ => "hours"
+    };
+
+    let minutes_label = match minutes {
+        1 => "minute",
+        _ => "minutes"
+    };
+
+    let seconds_label = match seconds {
+        1 => "second",
+        _ => "seconds"
+    };
+
+    let mut text = String::new();
+
+    if hours > 0 {
+        text.push_str(&format!("{value} {label}", value = hours, label = hours_label));
```

Comment:
If you import `std::fmt::Write` then you will be able to use a `write!` macro for formatting strings directly into a `text` without an extra String being allocated.

Reply:
doooope, thx!